### ISSUE TITLE
Incorrect type for OCaml services

### ISSUE THREAD

**Issue by vouillon** [ocsigen/eliom#364]
In `Eliom_services`, either the parameter of type `ocaml` or the last parameter of type `t` should be non-variant. Otherwise, the type of `Eliom_service.create_ocaml` is wrong : the OCaml type gets generalized. (See #223.)

**Comment by vasilisp** [comment]
Right. I thought I had fixed this. I pushed a fix to the `service-split-att` branch (cd0b7ab7c32796170aaf6bdc0d7912ab42ae00fd, #359).
